tsl/src/bgw_policy/compress_chunks_api.c

```diff
@@ -92,6 +97,21 @@ compress_chunks_add_policy(PG_FUNCTION_ARGS)
 						"enabled")));
 	}
 
+	role_tup = SearchSysCache1(AUTHOID, ObjectIdGetDatum(owner_id));
+	rform = (Form_pg_authid) GETSTRUCT(role_tup);
+
+	if (!rform->rolcanlogin)
+	{
+		ts_cache_release(hcache);
+		ReleaseSysCache(role_tup);
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION),
+				 errmsg("Unable to start compression background process for hypertabletable \"%s\"",
```

Comment:
```suggestion
				 errmsg("unable to start compression background process for hypertable \"%s\"",
```

I am wondering if this message shouldn't just say:

`"permission denied to create compression job for hypertable \"%s\""`

This gives a bit more context on why it failed and also seems more consistent with permission-related PG errors. (They usually are a variation of `permission denied to...`)


Note that these nits and suggestions apply to other places where similar errors occur as well.

Reply:
Went with `"permission denied to start ... background process"` after refactoring the code to a single location.